src/tokens/ERC20Gauges.sol

```diff
@@ -454,73 +320,40 @@ abstract contract ERC20Gauges is ERC20 {
     */
     function decrementGauges(
         address[] calldata gaugeList,
-        uint112[] calldata weights
-    ) public virtual returns (uint112 newUserWeight) {
+        uint256[] calldata weights
+    ) public virtual returns (uint256 newUserWeight) {
         uint256 size = gaugeList.length;
         require(weights.length == size, "ERC20Gauges: size mismatch");
 
         // store total in summary for batch update on user/global state
-        uint112 weightsSum;
-
-        uint32 currentCycle = _getGaugeCycleEnd();
+        uint256 weightsSum;
 
         // Update gauge specific state
         // All operations will revert on underflow, protecting against bad inputs
         for (uint256 i = 0; i < size; ) {
             address gauge = gaugeList[i];
-            uint112 weight = weights[i];
+            uint256 weight = weights[i];
             weightsSum += weight;
 
-            _decrementGaugeWeight(msg.sender, gauge, weight, currentCycle);
+            _decrementGaugeWeight(msg.sender, gauge, weight);
+            totalTypeWeight[gaugeType[gauge]] -= weight;
```

Comment:
this should be in the _decrementGaugeWeight function

Reply:
See previous comment on `_decrementGaugeWeight`